python/vespa/vespa/package.py

```diff
@@ -446,4 +457,8 @@ def deploy(self, disk_folder: str, container_memory: str = "4G"):
         deployment = self.container.exec_run(
             "bash -c '/opt/vespa/bin/vespa-deploy prepare /app/application && /opt/vespa/bin/vespa-deploy activate'"
         )
-        return deployment.output.decode("utf-8").split("\n")
+
+        return (
+            deployment.output.decode("utf-8").split("\n"),
+            Vespa(url="http://localhost", port=8080),
```

Comment:
Is it idiomatic to return a tuple with message and a handle? It seems odd, and how do you programatically see the difference between successes and failures here?

You should probably use self.local_port instead of directly referencing port 8080?

Reply:
1. Yes, you are right. The tuple seems odd. I suggest to store the deployment message as an attribute of the `Vespa` instance. What do you think?

2. About the difference between successes and failures: This info needs to be extracted from the deployment message. My plan is to improve this part of the code as I experience deployment failures as I do not have a full overview of the type of errors I can face at this moment.

3. I agree about `self.local_port`. I will make the change after we settle about the points above.

Reply:
I guess the idiomatic way would be to throw and exception on failure and return the handle on success? It can lead to a bit noisy code, but I think that's the way the Python standard libraries does it (e.g. opening a file failures). Then an error message could be encoded in the exception. Messages on success can be an attribute on the Vespa object.

Reply:
Yes, I agree. I will definitely throw an exception if a deployment failure occur. I will implement the exceptions as soon as I face a deployment failure. But I think we are agreeing that the deployment message will be included on the `Vespa` instance when it is returned (which will only happen if deployment goes through.)

Reply:
Also, I found a way to detect when vespa deployment fails and then the code raises RunTimeError exception with the deployment message showing what went wrong.

Reply:
Suddenly I wonder why port is a separate parameter to the constructor and not part of the URL. I'm slow to catch that :)

Reply:
Not sure why I chose that besides the fact that I guess url and port often comes as two different arguments in other libraries. I myself always wonder if port should be included or not when specifying urls. Do you suggest a change?

Reply:
But if we agree to unify url and port into one I can do so in another PR.

Reply:
In general the URL should only omit the port number if it's the default port number for the protocol specified in the URL. But yes, let's keep that for a different PR then :)